### ISSUE TITLE
Reference to `runtime-evaluated-base-classes` in the same file don't work

### ISSUE THREAD

**Issue by iurisilvio** [astral-sh/ruff#8250]
For some reason, `semantic.resolve_call_path(map_subscript(base))` resolves to `None` when the reference is in the same file.

https://github.com/astral-sh/ruff/blob/f5e850745cc9bcaeb308a3a139b4e86db58096aa/crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs#L43-L44

* A minimal code snippet that reproduces the bug.

In this example, `example1.MyBaseClass` is a runtime evaluated base class.

### example1.py

This triggers `TCH003` for `Sequence`.

```python
from __future__ import annotations

from collections.abc import Sequence  # TCH003


class MyBaseClass:
    pass


class Foo(MyBaseClass):
    foo: Sequence
```

### example2.py

This is `TCH` compliant.

```python
from __future__ import annotations

from collections.abc import Sequence

from example1 import MyBaseClass


class Foo(MyBaseClass):
    foo: Sequence
```

* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.

`ruff --select TCH example1.py example2.py`

* The current Ruff settings (any relevant sections from your `pyproject.toml`).

```
[tool.ruff.flake8-type-checking]
runtime-evaluated-base-classes = [
  "example1.MyBaseClass",
]
```

* The current Ruff version (`ruff --version`).

Main branch, currently 0.1.1.

**Comment by charliermarsh** [comment]
Thanks! Similar to (or the same as) https://github.com/astral-sh/ruff/issues/5486.

**Comment by charliermarsh** [comment]
I really appreciate that you dug into the code :)

**Comment by evanrittenhouse** [comment]
Feel free to assign this to me
